src/lib/pubkey/dsa/dsa.cpp

```diff
@@ -50,6 +64,7 @@ DSA_PrivateKey::DSA_PrivateKey(const AlgorithmIdentifier& alg_id,
                                const secure_vector<uint8_t>& key_bits) :
    DL_Scheme_PrivateKey(alg_id, key_bits, DL_Group_Format::ANSI_X9_57)
    {
+   BOTAN_ARG_CHECK(m_x.bits() <= m_group.q_bits(), "x must not be larger than q");
```

Comment:
I have a hunch that this check should better be constant time, because:

https://github.com/randombit/botan/blob/15228710faef1b85aaad0cad4f207b8b6dfcdf74/src/lib/math/numbertheory/monty_exp.cpp#L102-L105

Reply:
I think this is safe enough to not be constant time in this context. `bits` itself is constant time, and if we reject that means the key itself is just manifest invalid. So there is no real info leak.

We should check `x <= q` though rather than just the bit size, since `x == q` would be accepted, but isn't a valid DSA key.